components/event-bus/cmd/event-bus-publish-knative/handlers/publish.go

```diff
@@ -35,14 +34,16 @@ func WithRequestSizeLimiting(next http.HandlerFunc, limit int64) http.HandlerFun
 	}
 }
 
-func KnativePublishHandler(knativeLib *knative.KnativeLib, knativePublisher *publisher.KnativePublisher, tracer *trace.Tracer, opts *opts.Options) http.HandlerFunc {
+func KnativePublishHandler(knativeLib *knative.KnativeLib, knativePublisher *publisher.KnativePublisher,
+	tracer *trace.Tracer, opts *opts.Options) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		// init the trace span and context
 		traceSpan, traceContext := initTrace(r, tracer)
 		defer trace.FinishSpan(traceSpan)
 
 		// handle the knativeLib publish request
-		message, channelName, namespace, err := handleKnativePublishRequest(w, r, knativeLib, knativePublisher, traceContext, opts.MaxChannelNameLength)
+		message, channelName, namespace, err, status := handleKnativePublishRequest(w, r, knativeLib, knativePublisher,
```

Comment:
I see in many cases, `nil` is returned for two or more variables, Are all the cases verified to handle `nil` correctly?

Reply:
@abbi-gaurav This was old code, I just added a new return value. I believe checking this is out of the scope right now.
However, I believe this was covered by returning an error and checking the error in the beginning